### PR TITLE
Fix issues with parameters

### PR DESCRIPTION
**Pull request by buschbapti**
There was an issue with the way `Parameters(CartesianPose)` are converted. The only way I have found to properly solve that at the moment is to pass through an intermediate variable that ensure the conversion. This is not ideal at the moment but I need this fix for the state machine. I believe we should revisit all of this anyway to have a better way of handling those and avoid code duplication.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,8 @@ ARG ROS_VERSION=foxy
 
 FROM ros2_ws:${ROS_VERSION}
 
+RUN sudo curl http://repo.ros2.org/repos.key | sudo apt-key add -
```
Comment (buschbapti) [inline Dockerfile:5]: This line is temporary. It fixes a current issue with ROS having its key being revoked. I think we can keep it there for now and should remove it in later versions

**Review by domire8 — APPROVED** [review]
If this fixes your problems, I have nothing to complain

**Review by eeberhard — COMMENTED** [review]
Since [control libraries PR #145](https://github.com/epfl-lasa/control_libraries/pull/145) this morning, the CartesianPose `data()` method `overrides` instead of hides the base CartesianState `data()` method. 

Because `to_std_vector()` is only implemented in the base class and uses `this->data()`, overriding it should give the correct behaviour. But, it turns out hiding the base method also had the same effect (this->data() evaluated for CartesianPose::data()`, so actually nothing changed on the outside.

So, I'm not really sure what the problem is to prompt this PR. I definitely like the change illustrated below, but not so much the other intermediate variable stuff.

```
-  std::vector<double> vector_value = value.state_representation::CartesianPose::to_std_vector();
+ std::vector<double> vector_value = value.to_std_vector();
```

There are a few other places I've noticed where overrides are not always used properly, but in general I'm very pleased that we are chipping away with incremental improvements and small refactors.

**Comment by buschbapti** [comment]
There is indeed some polymorphism not resolving itself but it is due to a bad usage at the core. Basically, the problem lies in the DS lib that accepts a `Parameter<S>` as the attractor with `S` being a `CartesianState` or a `JointState`. Now at initilialization, this `Parameter<S>` is given a `CartesianPose` or a `JointPositions` but is still a `Parameter<State>`. Because there is a type variable in it, the function that decodes the parameters try to cast that as a `Parameter<Pose>` but it is in fact ill formed as it is in fact a `Pose` stored in a `Parameter<State>`. Therefore when using the functions `data` it actually calls the one of `State` not `Pose`.

I think the cleanest way would be to actually get rid of the conversion to `Parameter<Pose>` to only have `Parameter<State>`. This would remove a lot of code duplication but it comes with a drawback. It means that a parameter of pose or velocity or wrench will always be converted to a full state when exchanged in ROS meaning that you will need to fill all the other values that are not needed. I would be happy if there is a cleaner way that would resolve this.

File: source/packages/modulo_core/src/Cell.cpp [hunk 2]
```diff
@@ -56,7 +57,8 @@ void Cell::add_parameter<state_representation::CartesianPose>(const std::shared_
   std::string tprefix = (prefix != "") ? prefix + "_" : "";
   parameter->set_name(tprefix + parameter->get_name());
   this->parameters_.insert(std::make_pair(parameter->get_name(), parameter));
-  this->declare_parameter<std::vector<double>>(parameter->get_name(), parameter->get_value().state_representation::CartesianPose::to_std_vector());
```
[the remark this answers is not in the document]
Comment (buschbapti) [reply source/packages/modulo_core/src/Cell.cpp:59]: I think I tried the latest and it did not work neither but I am not sure. I will need to try again

**Review by eeberhard — APPROVED** [review]
I approve of this change for now so we can keep working, with a general refactor of this whole system planned for the future anyway.

**Comment by buschbapti** [comment]
@eeberhard I tried using `dynamic_cast` and it is indeed not working. Therefore I have reverted to use the only solution I know. As we discussed this is a perfect use case for `protobuff` and it would solve this kind of hack.

**Pull request merged**